### PR TITLE
Bayesian Model Averaged T-Test

### PR DESCRIPTION
**Pull request by FBartos**
Adds Bayesian Model Averaged T-Test analysis to JASP.

Currently requires:
RoBTT: `FBartos/RoBTT@JASP-updates2`
BayesTools: `FBartos/BayesTools@JASP-RoBTT-fixes`

### CODE REVIEW

**Review by vandenman — CHANGES_REQUESTED** [review]
The code looks mostly fine, although I haven't tested it yet in Jasp. I'm still not sure if we really want Stan and related packages this low in our dependency chain though. Before this PR "only" jaspMetaAnalysis, jaspMixedModels, jaspProphet, and jaspCochrane depend on Stan. After this PR that would also include jaspTTests, jaspAnova, jaspRegression, jaspEquivalenceTTests, and jaspSummaryStatistics. So most of the common modules would then depend on Stan, which is not a good development for our build process.

File: R/ttestbayesianmodelaveragedinternal.R [hunk 1]
```diff
@@ -0,0 +1,1177 @@
...
+    modelsSummary$addRows(list(
+      number             = fitSummary[["summary"]][i, "Model"],
+      distribution       = fitSummary[["summary"]][i, "Distribution"],
+      priorEffect        = fitSummary[["summary"]][i, "delta"],
+      priorHeterogeneity = fitSummary[["summary"]][i, "rho"],
+      priorOutliers      = fitSummary[["summary"]][i, "nu"],
+      priorProb          = fitSummary[["summary"]][i, "prior_prob"],
+      postProb           = fitSummary[["summary"]][i, "post_prob"],
+      marglik            = fitSummary[["summary"]][i, "marglik"],
+      BF                 = BayesTools::format_BF(bf[i], logBF = options[["bayesFactorType"]] == "LogBF10", BF01 = options[["bayesFactorType"]] == "BF01")
```
[the remark this answers is not in the document]
Comment (FBartos) [reply R/ttestbayesianmodelaveragedinternal.R:703]: It's a critical dependency of RoBTT, I can add it into imports too or rewrite it, depends what you think is better.

[the remark this answers is not in the document]
Comment (FBartos) [reply R/ttestbayesianmodelaveragedinternal.R:703]: btw, why isn't this function part of jaspBase?

```
.formatBf  <- function(BF, bayesFactorType){
  return(switch(
    bayesFactorType,
    "BF10"    = BF,
    "BF01"    = 1/BF,
    "LogBF10" = log(BF)
  ))
}
```
or I just can't find it?

File: inst/qml/common/ModelAveragedTTestPriors.qml [hunk 2]
```diff
@@ -0,0 +1,347 @@
...
+ColumnLayout
+{
+	spacing: 						0
+	property string componentType:	"Default type"
+
+	Label
+	{
+		text:					
+		{
+			if (componentType == "modelsEffect")
```
[the remark this answers is not in the document]
Comment (FBartos) [reply inst/qml/common/ModelAveragedTTestPriors.qml:33]: I changed it but I don't find it much simpler as there needs to be explicit break statements, i.e.,\

```
switch (componentType) 
		{
		  case "modelsEffect":				qsTr("Effect"); break;
		  case "modelsEffectNull":			qsTr("Effect (null)"); break;
		  case "modelsHeterogeneity":		qsTr("Heterogeneity (precision allocation)"); break;
		  case "modelsHeterogeneityNull":         qsTr("Heterogeneity (null)"); break;
		  case "modelsOutliers":		                qsTr("Outliers (degrees of freedom + 2)"); break;
		  case "modelsOutliersNull":	                qsTr("Outliers (null)"); break;
		}
```

**Pull request closed without merging**